src/tm2/tm-device.cpp

```diff
@@ -1428,6 +1447,7 @@ namespace librealsense
 
     void tm2_sensor::time_sync()
     {
+        device_to_host_ns = 0;
```

Comment:
This initialization should be in `class` definition to avoid a race with the thread that use this.